packages/react-components/react-tree/src/components/TreeItem/useTreeItem.tsx

```diff
@@ -148,88 +152,81 @@ export function useTreeItem_unstable<Value = string>(
     }
   });
 
-  const [isActionsVisible, setActionsVisible] = React.useState(false);
-  const showActions = useEventCallback((event: React.SyntheticEvent) => {
-    const isEventFromSubtree = subtreeRef.current && elementContains(subtreeRef.current, event.target as Node);
-    if (!isEventFromSubtree) {
+  const showActions = useEventCallback((event: React.FocusEvent) => {
+    const isTargetFromSubtree = Boolean(
+      subtreeRef.current && elementContains(subtreeRef.current, event.target as Node),
+    );
+    if (!isTargetFromSubtree) {
       setActionsVisible(true);
     }
   });
-  const hideActions = useEventCallback((event: React.SyntheticEvent) => {
-    const isEventFromSubtree = subtreeRef.current && elementContains(subtreeRef.current, event.target as Node);
-    if (!isEventFromSubtree) {
-      setActionsVisible(false);
+  const hideActions = useEventCallback((event: React.FocusEvent) => {
+    const isTargetFromSubtree = Boolean(
+      subtreeRef.current && elementContains(subtreeRef.current, event.target as Node),
+    );
+    const isRelatedTargetFromActions = Boolean(
+      actionsRef.current && elementContains(actionsRef.current, event.relatedTarget as Node),
+    );
+    if (isRelatedTargetFromActions) {
+      return setActionsVisible(true);
+    }
+    if (!isTargetFromSubtree) {
+      return setActionsVisible(false);
     }
   });
 
-  // Listens to focusout event on the document to ensure treeitem actions visibility on portal scenarios
-  // TODO: find a better way to ensure this behavior
+  const [isAsideVisible, setAsideVisible] = React.useState(false);
   React.useEffect(() => {
-    if (actionsRef.current) {
-      const handleFocusOut = (event: FocusEvent) => {
-        setActionsVisible(elementContains(actionsRef.current, event.relatedTarget as Node));
-      };
-      targetDocument?.addEventListener('focusout', handleFocusOut, { passive: true });
-      return () => {
-        targetDocument?.removeEventListener('focusout', handleFocusOut);
-      };
+    if (aside) {
+      setAsideVisible(actionsRef.current === null);
     }
-  }, [targetDocument]);
+  }, [isActionsVisible, aside]);
```

Comment:
Did not get what it's doing...

- We have `isActionsVisible` in dependencies, but don't use it
- `setAsideVisible` depends on DOM presence

Reply:
Yes! your doubt leads me to believe `isActionsVisible` is a bad name perhaps. 

`isActionsVisible` does not indicate that `actions` will be available in the DOM, it just indicates that on the `TreeItem` side it is the moment to show `actions`, but that only happens if `TreeItemActions` is actually declared.